### PR TITLE
Plugins: Unsigned chromium file should not invalidate signature for Renderer plugin

### PR DESCRIPTION
**Pull request by wbrowne**
This change is making plugin signing responsible for ignoring this generated plugin file during validating, rather than relying on the rendering service manually removing the file on shutdown.

Long term, we'd like to address the core issue of Chromium generating this file (Windows only), but for now this work is part of a bigger current effort to un-expose `PluginDir` as a property of a plugin.

### CODE REVIEW

File: pkg/plugins/manager/signature/manifest.go [hunk 1]
```diff
@@ -177,6 +180,9 @@ func Calculate(mlog log.Logger, plugin *plugins.Plugin) (plugins.Signature, erro
 	var unsignedFiles []string
 	for _, f := range pluginFiles {
 		if _, exists := manifestFiles[f]; !exists {
+			if runningWindows && plugin.IsRenderer() && f == "chrome-win/debug.log" {
```
Comment (leventebalogh) [inline pkg/plugins/manager/signature/manifest.go:183]: Even though the tests serve as an explanation, would it make sense to add a comment above this line as well explaining why it is needed? (Asking as I have a 6th sense that new devs might don't understand it immediately). E.g.: `"Ignoring unsigned Chromium debug.log so it doesn't invalidate the signature for Renderer plugin running on Windows"`

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Review by marefr — APPROVED** [review]
LGTM, but would be nice if the debug logs rather was output:ed to stdout/stderr as it works for Mac/Linux. Guess it's not possible to configure that with puppeteer/chromium?

**Comment by wbrowne** [comment]
@marefr I should have mentioned I tried some experiments with different image renderer builds using different flags (for example https://peter.sh/experiments/chromium-command-line-switches/#disable-logging). Unfortunately nothing seemed to work.

However, `--disable-gpu` was added in a later version to the renderer and it seemed to stop generating the file. But that could just be preventing those particular errors from being logged and not any others.

**Comment by marefr** [comment]
> @marefr I should have mentioned I tried some experiments with different image renderer builds using different flags (for example https://peter.sh/experiments/chromium-command-line-switches/#disable-logging). Unfortunately nothing seemed to work.
> 
> However, `--disable-gpu` was added in a later version to the renderer and it seemed to stop generating the file. But that could just be preventing those particular errors from being logged and not any others.

That's what I suspected so then I guess this is the way currently :+1:

File: pkg/plugins/manager/signature/manifest.go (hunk 1 above)
Comment (marefr) [reply pkg/plugins/manager/signature/manifest.go:183]: Could this file be made executable? If that's the case we should probably fail to load the plugin

Comment (wbrowne) [reply pkg/plugins/manager/signature/manifest.go:183]: My Windows knowledge is lacking here, but maybe that's possible? We could fail this scenario if we have a valid check for Windows, but I'm not sure Go provides this out of the box. Any ideas?

Comment (wbrowne) [reply pkg/plugins/manager/signature/manifest.go]: @jmatosgrafana wanted to loop you in here to weigh in if you don't mind!

Comment (jmatosgrafana) [reply pkg/plugins/manager/signature/manifest.go]: @wbrowne LGTM from a security point of view as long as the plugin never tries to execute `debug.log` (a bit like `MANIFEST.txt`)

**Pull request merged**